### PR TITLE
Add basic concurrency test for ray client

### PR DESCRIPTION
**Pull request by ericl**

### CODE REVIEW

**Comment by richardliaw** [comment]
nice! let me try this out.

**Comment by richardliaw** [comment]
does it make sense to test this?

**Comment by ericl** [comment]
Actually, I realized this is not needed since the client impl is actually thread-safe (added a simple test). I think your original issue is unrelated.

**Review by AmeerHajAli — APPROVED** [review]
LGTM. Not sure though how it "fixes" the other issue (instead of verifying it works).

**Comment by ericl** [comment]
Sorry, it's actually not fixing the issue (fixed the description). This is just adding another test to avoid regressions on this front.

File: python/ray/tests/test_client.py [hunk 1]
```diff
@@ -10,6 +10,39 @@
...
+
+            def run(self):
+                ray.get(block.remote())
+
+        b = Blocker()
+        b.start()
+        time.sleep(1)
+
+        # Can concurrently execute the get.
+        assert ray.get(fast.remote(), timeout=5) == "ok"
```
Comment (ijrsvt) [inline python/ray/tests/test_client.py:43]: OOC, will this introduce flakiness in an over-loaded test environment?

**Comment by ericl** [comment]
The timeout is 5 seconds, which should be pretty generous... depends on how
overloaded the servers are. I can increase it to say 15 if there is a
concern?

**Comment by ijrsvt** [comment]
@ericl 5 is probably fine, we can always bump in the future

**Review by ijrsvt — APPROVED** [review]
LGTM

**Pull request merged**